dll/win32/shlwapi/propbag.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * PROJECT:     ReactOS Shell
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Implement shell property bags
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#define _ATL_NO_EXCEPTIONS
+#include "precomp.h"
+#include <shlwapi.h>
+#include <shlwapi_undoc.h>
+#include <atlstr.h>         // for CStringW
+#include <atlsimpcoll.h>    // for CSimpleMap
+#include <atlcomcli.h>      // for CComVariant
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+class CBasePropertyBag
+    : public IPropertyBag
+#if (_WIN32_WINNT < _WIN32_WINNT_VISTA)
+    , public IPropertyBag2
```

Comment:
Oh, that's interesting. So (for once!) they merged both interfaces into a single one in Vista+ ? or...?

Reply:
I didn't test it on Win7 nor Win8.
Vista is my expectation.

Reply:
You should try checking on Win7 and 10 out of curiosity / documentation.

Reply:
Win10 is tested on my PC.